### PR TITLE
Hds 1168 fix dateinput perf

### PR DESCRIPTION
**Pull request by minevala**
## Description
- Upgrade popperjs libraries
- Render DatePicker only when it is opened
- Update DatePicker snapshot tests

## Related Issues
- https://helsinkisolutionoffice.atlassian.net/browse/HDS-1168
- https://helsinkisolutionoffice.atlassian.net/browse/HDS-1494

## Motivation and Context
- DateInput renders DatePicker always, even if it is not opened. This might cause some performance issues because DatePicker is rendered when the overall form state changes.

## How Has This Been Tested?
- locally on dev machine
- functional component tests

[Demo](https://city-of-helsinki.github.io/hds-demo/hds-1168-fix-dateinput-perf/?path=/story/components-dateinput--default)

### CODE REVIEW

**Comment by minevala** [comment]
> Warning from Tooltip [tests](https://github.com/City-of-Helsinki/helsinki-design-system/actions/runs/3650303640/jobs/6166114007#step:8:38):
> 
> > Warning: flushSync was called from inside a lifecycle method. React cannot flush when React is already rendering. Consider moving this call to a scheduler task or micro task.
> 
> Needs some investigation

I had to downgrade react-popper because the latest version is not probably compatible with React 17

File: packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx [hunk 1]
```diff
@@ -300,11 +303,18 @@ export const DatePicker = (providedProps: DayPickerProps) => {
 
   const currentMonthAvailableDays: number[] = currentMonthAvailableDates.map((date) => date.getDate());
 
+  const onPopperRender = React.useCallback(() => {
```
Comment (minevala) [inline packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx:306]: onPopperReady

File: packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx [hunk 2]
```diff
@@ -300,11 +303,18 @@ export const DatePicker = (providedProps: DayPickerProps) => {
...
+      setIsReadyToShow(true);
+    }
+  }, [isReadyToShow, setIsReadyToShow]);
+
   // Initialize Popper.js
   const { styles: datePickerPopperStyles, attributes: datePickerPopperAttributes } = usePopper(
     inputRef.current,
     pickerWrapperRef.current,
     {
+      onFirstUpdate: onPopperRender,
```
Comment (minevala) [inline packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx:317]: Try Popper lifecycle methods

Comment (minevala) [inline packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx:317]: Add explanatory comment

File: packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx [hunk 3]
```diff
@@ -76,6 +76,8 @@ export const DatePicker = (providedProps: DayPickerProps) => {
    */
   const [selectedDate, setSelectedDate] = useState<Date>(selected || null);
 
+  const [isReadyToShow, setIsReadyToShow] = useState<boolean>(false);
```
[the remark this answers is not in the document]
Comment (minevala) [reply packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx]: - [x] rename constants

File: packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx (hunk 2 above)
Comment (minevala) [reply packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx]: - [x] use lifecycle methods

Comment (minevala) [reply packages/react/src/components/dateInput/components/datePicker/DatePicker.tsx]: - [x] add comment

**Review by harriplappalainen — APPROVED** [review]
Good fixes to tests and needless rendering 🏅

**Pull request merged**